rs/bitcoin/ckbtc/minter/src/updates/tests.rs

```diff
@@ -0,0 +1,294 @@
+            UpdateBalanceArgs {
+                owner: Some(account.owner),
+                subaccount: account.subaccount,
+            },
+            &runtime,
+        )
+        .await;
+
+        assert_eq!(result, Ok(vec![UtxoStatus::Tainted(quarantined_utxo)]));
+        assert_has_no_new_events(&events_before);
+    }
+
+    #[tokio::test]
+    async fn should_mint_reevaluated_tainted_utxo() {
+        init_state_with_ecdsa_public_key();
+        let account = ledger_account();
+        let quarantined_utxo = quarantined_utxo();
+        register_utxo_checked(&quarantined_utxo, UtxoCheckStatus::Tainted);
+        let kyt_fee = read_state(|s| s.kyt_fee);
+        let minted_amount = quarantined_utxo.value - kyt_fee;
+        let events_before: Vec<_> = storage::events().collect();
+
+        let mut runtime = MockCanisterRuntime::new();
+        mock_get_utxos_for_account(&mut runtime, account, vec![quarantined_utxo.clone()]);
+        expect_check_transaction_returning(
+            &mut runtime,
+            quarantined_utxo.clone(),
+            CheckTransactionResponse::Passed,
+        );
+        runtime
+            .expect_mint_ckbtc()
+            .times(1)
+            .withf(move |amount, account_, _memo| amount == &minted_amount && account_ == &account)
+            .return_const(Ok(1));
+        mock_schedule_now_process_logic(&mut runtime);
+        let result = update_balance(
+            UpdateBalanceArgs {
+                owner: Some(account.owner),
+                subaccount: account.subaccount,
+            },
+            &runtime,
+        )
+        .await;
+
+        assert!(!read_state(|s| s.has_ignored_utxo(&quarantined_utxo)));
+        assert_eq!(
+            result,
+            Ok(vec![UtxoStatus::Minted {
+                block_index: 1,
+                minted_amount,
+                utxo: quarantined_utxo.clone(),
+            }])
+        );
+        assert_has_new_events(
+            &events_before,
+            &[
+                checked_utxo_event(quarantined_utxo.clone(), true),
+                Event::ReceivedUtxos {
+                    mint_txid: Some(1),
+                    to_account: account,
+                    utxos: vec![quarantined_utxo],
+                },
+            ],
+        );
+    }
+
+    fn init_state_with_ecdsa_public_key() {
+        use crate::lifecycle::init::InitArgs;
+        use ic_base_types::CanisterId;
+        init_state(InitArgs {
+            new_kyt_principal: Some(CanisterId::unchecked_from_principal(KYT_CANISTER_ID.into())),
+            ..init_args()
+        });
+        mutate_state(|s| s.ecdsa_public_key = Some(ecdsa_public_key()))
+    }
+
+    fn register_utxo_checked(utxo: &Utxo, status: UtxoCheckStatus) {
+        mutate_state(|s| {
+            audit::mark_utxo_checked(s, utxo, None, status, None);
+        });
+    }
+
+    fn expect_bitcoing_get_utxos_returning(runtime: &mut MockCanisterRuntime, utxos: Vec<Utxo>) {
```

Comment:
`expect_bitcoing_get_utxos_returning` -> `expect_bitcoin_get_utxos_returning`

Reply:
It took me surprisingly long to find the unexpected `g` 🙈. Done.